libbpf-rs/src/map.rs

```diff
@@ -1282,6 +1356,129 @@ impl Iterator for MapKeyIter<'_> {
     }
 }
 
+/// An iterator over batches of key value pairs of a BPF map.
+#[derive(Debug)]
+pub struct BatchedMapIter<'map> {
+    map_fd: BorrowedFd<'map>,
+    delete: bool,
+    count: usize,
+    key_size: usize,
+    value_size: usize,
+    keys: Vec<u8>,
+    values: Vec<u8>,
+    prev: Option<Vec<u8>>,
+    next: Vec<u8>,
+    batch_opts: libbpf_sys::bpf_map_batch_opts,
+    index: Option<usize>,
+}
+
+impl<'map> BatchedMapIter<'map> {
+    fn new(
+        map_fd: BorrowedFd<'map>,
+        count: u32,
+        key_size: u32,
+        value_size: u32,
+        batch_opts: libbpf_sys::bpf_map_batch_opts,
+        delete: bool,
+    ) -> Self {
+        Self {
+            map_fd,
+            delete,
+            count: count as usize,
+            key_size: key_size as usize,
+            value_size: value_size as usize,
+            keys: vec![0; (count * key_size) as usize],
+            values: vec![0; (count * value_size) as usize],
+            prev: None,
+            next: vec![0; key_size as usize],
+            batch_opts,
+            index: None,
+        }
+    }
+
+    fn lookup_next_batch(&mut self) {
+        let prev = self.prev.as_ref().map_or(ptr::null(), |p| p.as_ptr());
+        let mut count = self.count as u32;
+
+        let ret = unsafe {
+            if self.delete {
+                libbpf_sys::bpf_map_lookup_and_delete_batch(
+                    self.map_fd.as_raw_fd(),
+                    prev as _,
```

Comment:
FWIW, I think this cast silently masks undefined behavior (mutation of a non-`mut` variable), which you could have easily spotted had you followed the suggestion of using `cast` thoroughly, because that's exactly what it's meant to catch.